src/QuartoPlayerAgent.java

```diff
@@ -143,31 +128,26 @@ protected String moveSelectionAlgorithm(int pieceID)
     {
         System.out.println("Entered move selection");
 
-        QuartoPiece givenPiece = this.curState.board.getPiece(pieceID);
         QuartoGameState prevState = this.curState;
+        QuartoPiece givenPiece = this.curState.board.getPiece(pieceID);
+
+        QuartoGameTransition quartoGameTransition = null;
 
-        // Set this.curState
-        QuartoGameTransition quartoGameTransition = this.curState.transitions.get(givenPiece.binaryStringRepresentation() + ":" +
-                                                    minisChosenSquare[0] + "," + minisChosenSquare[1]);
+        if (minisPieceID != -1) {
+            quartoGameTransition = prevState.transitions.get(prevState.bestTransition.placedPiece.binaryStringRepresentation() + ":" +
```

Comment:
If this is the first move (i.e. we are placing the first piece) prevState won't have a bestTransition